urlparse4/cgurl.pyx

```diff
@@ -63,14 +68,94 @@ cdef bytes build_netloc(bytes url, Parsed parsed):
         raise ValueError
 
 
-def unicode_handling(str):
+cdef bytes unicode_handling(str):
     cdef bytes bytes_str
     if isinstance(str, unicode):
         bytes_str = <bytes>(<unicode>str).encode('utf8')
     else:
         bytes_str = str
     return bytes_str
 
+cdef void parse_url(bytes url, Component url_scheme, Parsed * parsed):
+    if CompareSchemeComponent(url, url_scheme, kFileScheme):
+        ParseFileURL(url, len(url), parsed)
+    elif CompareSchemeComponent(url, url_scheme, kFileSystemScheme):
+        ParseFileSystemURL(url, len(url), parsed)
+    elif IsStandard(url, url_scheme):
+        ParseStandardURL(url, len(url), parsed)
+    elif CompareSchemeComponent(url, url_scheme, kMailToScheme):
+        """
+        Discuss: Is this correct?
```

Comment:
the logic looks good to me, kMailToScheme -> ParseMailtoURL, what would you like to discuss?

Reply:
I will create an issue to discuss about this. I believe there's a test that urlparse4 did not pass because of this 😄